### ISSUE TITLE
[BUG] check-unused-code quirks (digits in filenames, macOS)

### ISSUE THREAD

**Issue by furaiev** [dart-code-checker/dart-code-metrics#855]
Strange behavior that depends on the filename.
When the filename contains digits, the `check-unused-code` reports code as unused (on macOS only).

* **Dart code metrics version:** 4.15.2
* **Dart sdk version:** 2.17.0

```dart
// main.dart

part 'part_of_test_class.dart';
part 'part_of_test_class_2.dart';

void main() {
  _PartOfTestClass();
  _PartOfTestClassWithNumberInFileName();
}

// part_of_test_class.dart

part of 'main.dart';

class _PartOfTestClass {
  String get testString => 'testString';
}


// part_of_test_class_2.dart

part of 'main.dart';

class _PartOfTestClassWithNumberInFileName {
  String get testString => 'testString';
}

```

**macOS result:**
```
lib/part_of_test_class_2.dart:
    ⚠ unused class _PartOfTestClassWithNumberInFileName
      at lib/part_of_test_class_2.dart:3:1

✖ total unused code (classes, functions, variables, extensions, enums, mixins and type aliases) - 1
```

**Linux result:**
```
✔ no unused code found!
```

**min reproduce repo:** https://github.com/furaiev/code_metrics_proof

**macOS codemagic result (see Pre-test script logs):** https://codemagic.io/app/6283afc607752fbaa9523008/build/6284a37f96304c0456b891ab

**Linux codemagic result (see Pre-test script logs):** https://codemagic.io/app/6283afc607752fbaa9523008/build/6284a45d96304c402203395b

**Comment by incendial** [comment]
Interesting, this looks like an analyzer issue, not ours 🤔 

@furaiev do you have the word `test` in our original files?

**Comment by incendial** [comment]
Asking, coz removing the word (actually, naming any other way) produces 0 issues

**Comment by furaiev** [comment]
@incendial removed test from names

```
lib/part_of_class_2.dart:
    ⚠ unused class _PartOfClassWithNumberInFileName
      at lib/part_of_class_2.dart:3:1

✖ total unused code (classes, functions, variables, extensions, enums, mixins and type aliases) - 1
```

https://codemagic.io/app/6283afc607752fbaa9523008/build/628a6929d44138ebc3ba6a42

**Comment by incendial** [comment]
👍  I'll report this issue to the Dart team, looks like there is some internal problem with how element model is being built. Let's see, what they tell.

**Comment by incendial** [comment]
Reported here: https://github.com/dart-lang/sdk/issues/49182

**Comment by incendial** [comment]
Available in 4.16.0! 🚀

**Comment by furaiev** [comment]
@incendial from the first sight it works! Thanks!
